File: app/src/main/java/com/duy/pascal/backend/exceptions/grouping/StrayCharacterException.java
package com.duy.pascal.backend.exceptions.grouping;

import com.duy.pascal.backend.linenumber.LineInfo;

public class StrayCharacterException extends com.duy.pascal.backend.exceptions.grouping.GroupingException {

    public StrayCharacterException(LineInfo line, char character) {
        super(line, "Stray character in program: " + character
                + "\nChar code " + (int) character);
    }

}

File: libCompiler/src/main/java/com/duy/pascal/backend/parse_exception/grouping/StrayCharacterException.kt
package com.duy.pascal.backend.parse_exception.grouping

import com.duy.pascal.backend.linenumber.LineInfo

class StrayCharacterException : GroupingException {
    var charCode: Char

    constructor(line: LineInfo, charCode: Char) : super(line,
            "Stray character in program: $charCode\nChar code ${charCode.toInt()}") {
        this.charCode = charCode
    }
}

